src/engine/strat_engine/range_alloc.rs

```diff
@@ -57,44 +97,58 @@ impl RangeAllocator {
         for &(off, len) in to_free {
             self.check_for_overflow(off, len);
 
-            for val in *off..*off + *len {
-                let removed = self.used.remove(&val);
-                if removed == false {
-                    panic!(format!("tried to remove value {} not present in RangedSet", val));
+            let (prev_off, prev_len) = self.used
+                .range((Unbounded, Included(off)))
+                .rev()
+                .next()
+                .map(|(k, v)| (*k, *v))
+                .expect("must match an existing range");
```

Comment:
Let's discuss API issues separate from this PR, since this PR doesn't change the API.

I still don't understand the objection here. The old code panic()ed if you tried to remove a non-present range, the new code does too -- the expect fails and panic()s.